src/fastpath/fastpath.c

```diff
@@ -202,8 +210,10 @@ void NORETURN fastpath_signal(word_t cptr, word_t msgInfo)
 #endif
 
     // TODO: Needed?
-    /* Try and wake up threads - this makes everything fail */
-    // awaken();
+    /* This is the check used in awaken() - if we have to do this, we should go to the slowpath */
+    if (unlikely(NODE_STATE(ksReleaseHead) != NULL && refill_ready(NODE_STATE(ksReleaseHead)->tcbSchedContext))) {
+        slowpath(SysSend);
```

Comment:
I don't think so, I think this would have been true before this commit since the code for actually doing the signal would run regardless of whether the direct context switch worked. In the most recent commit, I moved this logic so that it is inside the `if (NODE_STATE(ksCurThread)->tcbPriority >= dest->tcbPriority)` statement. Since that state change only executes if we are signalling a lower prio thread and this check should only be reached if we are signalling to a higher prio thread, I don't think any state is changed.

Reply:
Cool, yes, you're right that is fine.